### PR TITLE
parsers: add nickel parser

### PR DESCRIPTION
**Pull request by jlesquembre**
Add nickel parser: https://github.com/nickel-lang/tree-sitter-nickel

Nickel lang: https://nickel-lang.org/

The highlights.scm file is from: https://github.com/nickel-lang/tree-sitter-nickel/blob/main/queries/highlights.scm (I asked the author and he is ok with upstreaming it)

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1213,6 +1213,16 @@ list.gitignore = {
   maintainers = { "@theHamsta" },
 }
 
+list.nickel = {
+  install_info = {
+    url = "https://github.com/nickel-lang/tree-sitter-nickel",
+    files = { "src/parser.c", "src/scanner.cc" },
+    branch = "main",
+    generate_requires_npm = false,
+    requires_generate_from_grammar = false,
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1222]:
```suggestion
```
No need to specify the defaults.

File: queries/nickel/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,64 @@
+(comment) @comment
```
[the remark this answers is not in the document]
Comment (jlesquembre) [reply queries/nickel/highlights.scm:1]: I don't think so, but I'm not sure, I just took it from the grammar repo: https://github.com/nickel-lang/tree-sitter-nickel

**Comment by jlesquembre** [comment]
@clason thanks for the review, I think this should fix the error: https://github.com/nvim-treesitter/nvim-treesitter/pull/3506/commits/4304885945742f55306f6db532262ecf7f5949aa

File: queries/nickel/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,64 @@
+(comment) @comment @spell
+
+[
+  "if"
+  "then"
+  "else"
+  "forall"
```
Comment (theHamsta) [inline queries/nickel/highlights.scm:7]: `@repeat`?

File: queries/nickel/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,64 @@
+(comment) @comment @spell
+
+[
+  "if"
+  "then"
+  "else"
```
Comment (theHamsta) [inline queries/nickel/highlights.scm:6]: `@conditional` for if/else/then

Comment (theHamsta) [reply queries/nickel/highlights.scm:6]: you already have the conditional highlight. I don't think keyword is necessary then

File: queries/nickel/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+(interpolation_end) @punctuation.bracket
+
+(record_field) @field
+
+(builtin) @function.builtin
+
+(fun_expr pats:
+  (pattern id:
+    (ident) @parameter
+  )+
```
Comment (theHamsta) [inline queries/nickel/highlights.scm:59]: the `+` will have no effect here, but it also does not harm

File: queries/nickel/highlights.scm (hunk 3 above)
Comment (jlesquembre) [reply queries/nickel/highlights.scm:4]: Not sure, `forall` is used to define polymorphic types: https://nickel-lang.org/user-manual/typing/#type-polymorphism
It's not really a loop

File: queries/nickel/highlights.scm (hunk 4 above)
Comment (jlesquembre) [reply queries/nickel/highlights.scm]: Probably not, removed

**Comment by theHamsta** [comment]
Thank you!

**Pull request merged**